plugins/hw_wallet/qt.py

```diff
@@ -228,3 +228,7 @@ def choose_device(self, window, keystore):
 
     def show_settings_dialog(self, window, keystore):
         device_id = self.choose_device(window, keystore)
+
+        # Check if the device is paired and retry the pairing if not
+        if self.device_manager().xpub_by_id(device_id) == None:
+            keystore.thread.add(partial(self.get_client, keystore))
```

Comment:
This code (not your code) is so terrible to my OOP sensibilities.  There should be a default implementation of get_client that does nothing (perhaps even `self.print_error("get_client unimplemented")`. This way it's pretty clear for plugin creators what needs to be implemented.

Clearly all of the hw_wallet code expects this method to exist for a class inheriting from HW_Plugin ... So reading the code a programmer should be able to tell quickly what methods he needs to implement. 

Just a comment here.. not your fault.

Yeah this seems ok to me.  It sort of bugs me that rapid UI actions can trigger multiple ones of these get_client calls to happen in the keystore thread.. but this happens all over this codebase, really.

Reply:
I think I should check there if another get_client call is already running.